### ISSUE TITLE
custom_domain_verification_id output not being written to state file for azurerm_windows_function_app and azurerm_windows_function_app_slot resources

### ISSUE THREAD

**Issue by bdorplatt** [hashicorp/terraform-provider-azurerm#17042]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.0

### AzureRM Provider Version

3.8.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_function_app, azurerm_windows_function_app_slot

### Terraform Configuration Files

```hcl
output "fa_custom_domain_verification_id" {
  description = "Custom Domain name verification ID"
  value       = azurerm_windows_function_app.function_app.custom_domain_verification_id
  sensitive   = true
```

### Debug Output/Panic Output

```shell
State File:

 "outputs": {
    "fa_custom_domain_verification_id": {
      "value": "",
      "type": "string",
      "sensitive": true
    },
```

### Expected Behaviour

The Custom Domain Verification ID should be output to the state file for use as inputs for other modules.

### Actual Behaviour

A blank output is written to the state file.

This works correctly for the resouce: azurerm_windows_web_app 

Looking at the code, it appears there are discrepancies in what is written to state between the **azurerm_windows_web_app** and the **azurerm_windows_function_app** resources. Notably, **CustomDomainVerificationId** is missing from the section "state := WindowsFunctionAppModel" for  **azurerm_windows_function_app** 

**azurerm_windows_function_app**
Starting at line 580, **CustomDomainVerificationId** isn't present
[https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/appservice/windows_function_app_resource.go](url)

**azurerm_windows_web_app**
On line 500, there is an entry for:
**CustomDomainVerificationId**
[https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/appservice/windows_web_app_resource.go](url)

It appears these are missing for **azurerm_linux_function_app** and **azurerm_linux_function_app_slot** as well.

### Steps to Reproduce

Apply code for any function app or function app slot with custom_domain_verification_id as an output

**Comment by SergiyDivnych** [comment]
Hi, I have the same issue

**Comment by SergiyDivnych** [comment]
Works now in azurerm provider v3.10.0

**Comment by jackofallops** [comment]
Closed by #17183
